libcodechecker/cmd/cmd_line_client.py

```diff
@@ -321,7 +327,66 @@ def getDiffReportDir(client, baseids, report_dir, diff_type):
                     filtered_reports.append(result)
         return filtered_reports
 
-    def printReports(client, reports, output_format):
+    def cache_report_file(file_cache, file_id):
+        if file_id not in file_cache:
+            source = client.getSourceFileData(file_id, True,
+                                              ttypes.Encoding.BASE64)
+            file_content = base64.b64decode(source.fileContent)
+            file_cache[file_id] = {'id': file_id,
+                                   'path': source.filePath,
+                                   'content': file_content}
+
+        return file_cache[file_id]
+
+    def get_report_data(client, reports, file_cache):
+        file_sources = {}
+        report_data = []
+
+        for report in reports:
+            file_sources[report.fileId] = cache_report_file(file_cache,
+                                                            report.fileId)
+            details = client.getReportDetails(report.reportId)
+            events = []
+            for index, event in enumerate(details.pathEvents):
+                file_sources[event.fileId] = cache_report_file(file_cache,
+                                                               event.fileId)
+                events.append({'line': event.startLine,
+                               'col': event.startCol,
+                               'file': event.fileId,
+                               'msg': event.msg,
+                               'step': index + 1})
+            report_data.append(events)
+
+        return {'files': file_sources,
+                'reports': report_data}
+
+    def report_to_html(client, reports, output_dir):
+        html_builder = PlistToHtml.HtmlBuilder(context.path_plist_to_html_dist)
+
+        file_report_map = defaultdict(list)
+        for report in reports:
+            file_report_map[report.fileId].append(report)
+
+        file_cache = {}
+        for file_id, file_reports in file_report_map.iteritems():
+            checked_file = file_reports[0].checkedFile
+            filename = os.path.basename(checked_file)
+
+            report_data = get_report_data(client, file_reports, file_cache)
+
+            output_path = os.path.join(output_dir,
+                                       filename + '_' + str(file_id) + '.html')
+            html_builder.create(output_path, report_data)
+            print('Html file was generated for file {0}: {1}'.format(
+                checked_file, output_path))
+
+    def printReports(client, reports, output_format, diff_type):
```

Comment:
Parts of this file contains functions that break this convention, but Python uses `print_reports` conventionally for function names. Let's not introduce more unneccessary breaking.

Reply:
I didn't break the python coding convention. The convention was broken in the #723 pull request. I created a separate issue to fix this (#1070).